frontend/public/components/utils/copy-to-clipboard.jsx

```diff
@@ -1,15 +1,80 @@
 import * as React from 'react';
 
 import { CopyToClipboard as CTC } from 'react-copy-to-clipboard';
+import { Tooltip } from 'react-lightweight-tooltip';
 
-export const CopyToClipboard = ({value, visibleValue = value}) => <React.Fragment>
-  <div className="co-copy-to-clipboard">
-    <pre className="co-pre-wrap co-copy-to-clipboard__text">{visibleValue}</pre>
-    <CTC text={value}>
-      <button className="btn btn-default co-copy-to-clipboard__btn" type="button">
-        <i className="fa fa-clipboard" aria-hidden="true"></i>
-        <span className="sr-only">Copy to Clipboard</span>
-      </button>
-    </CTC>
-  </div>
-</React.Fragment>;
+export class CopyToClipboard extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {copied: false};
+    this.showCopied = this.showCopied.bind(this);
+    this.showDefault = this.showDefault.bind(this);
+  }
+
+  showCopied() {
+    this.setState({copied: true});
+  }
+
+  showDefault() {
+    this.setState({copied: false});
+  }
+
+  render() {
+    const overrides = {
```

Comment:
Consider `Object.freeze` as used in tooltip.jsx

https://github.com/openshift/console/blob/master/frontend/public/components/utils/tooltip.jsx#L4

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze